packages/central-server/src/apiV2/GETHandler/helpers.js

```diff
@@ -86,6 +86,35 @@ const getForeignKeyColumnName = foreignTable => {
   return exceptions[foreignTable] || `${foreignTable}_id`;
 };
 
+const addJoin = (
+  recordType,
+  recordTypesInQuery,
+  multiJoin,
+  baseRecordType,
+  customJoinConditions,
+  joinType,
```

Comment:
Although js will allow it, it's an anti-pattern to mutate (change) function parameters. This is a really good learning opportunity - here you are passing in `multiJoin` and expecting the function to add more joins to it, but you should _never do this_, or you end up in all sorts of trouble. Instead, you should make a _deep_ clone of the object, mutate that, and return it for the caller to use instead of its original object. If you allow functions to mutate things inside them, all of a sudden you lose predictability, and can't trust what invisible side effects might be going on. 
Can you please share this learning with Chao too, it's not obvious when you first start js but important to avoid getting in a pickle.